src/main/java/com/serenitydojo/fruitmarket/Catalog.java

```
@@ -0,0 +1,37 @@
+package com.serenitydojo.fruitmarket;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Catalog {
+
+    private Map<Fruits, Double> fruitCatalog;
+
+    public Catalog(Map<Fruits, Double> fruitCatalog) {
+        this.fruitCatalog = fruitCatalog;
+    }
+
+    public List<String> getAvailableFruits() {
+        return fruitCatalog.keySet()
+                .stream()
+                .map(Enum::name)
+                .sorted()
+                .collect(Collectors.toList());
+    }
+
+    public double getPriceOf(Fruits fruit) {
+        if(fruitCatalog.containsKey(fruit)) {
+            return fruitCatalog.get(fruit);
+        }
+        throw new FruitNotAvailableException(fruit.name() + " is not available");
+    }
+
+    public void setPriceOf(Fruits fruit, Double price) {
+        if(fruitCatalog.containsKey(fruit)) {
+            fruitCatalog.replace(fruit, price);
+        }
```

Should you not also throw an exception if the fruit is not available in the catalog?

---

I added the code to throw and exception when the fruit is not available. 
`    public void setPriceOf(Fruits fruit, Double price) {
        if (fruitCatalog.containsKey(fruit)) {
            fruitCatalog.replace(fruit, price);
        } else {
            throw new FruitNotAvailableException(fruit.name() + " is not available");
        }
    }`

I also included a new test to check that behaviour in the CatalogTest 
`@Test(expected = FruitNotAvailableException.class)
    public void update_price_of_a_fruit_not_in_catalog() {

        catalog.setPriceOf(Pear, 3.45);
    }`